src/twisted/words/protocols/jabber/jid.py

```diff
@@ -141,16 +143,19 @@ class JID:
     dictionaries.
     """
 
-    def __init__(self, str=None, tuple=None):
-        if not (str or tuple):
-            raise RuntimeError(
-                "You must provide a value for either 'str' or " "'tuple' arguments."
-            )
-
+    def __init__(
+        self,
+        str: Union[str, None] = None,
+        tuple: Union[Tuple[str, str, str], None] = None,
+    ):
         if str:
             user, host, res = parse(str)
-        else:
+        elif tuple:
             user, host, res = prep(*tuple)
+        else:
```

Comment:
I think that this  change is ok.

Is Twisted jabber still used by people in PROD ?

Reply:
change was required to appease mypy for the `*tuple` otherwise it was optional in the `if not str:` branch